main.js

```diff
@@ -203,7 +204,7 @@ ipcMain.handle("open-dialog", async (e, title) => {
     return await dialog.showOpenDialog(options);
 });
 
-ipcMain.on("settings:set", (e, setting, argument) => {
+ipcMain.handle("settings:set", (e, setting, argument) => {
```

Comment:
Handle is for messages that act like functions which return values and are sent using the "invoke"method on ipcRenderer. On is for messages which act like procedures and do not return anything and are sent using the "send" method on ipcRenderer. You have completely removed the ability of users to set settings by changing this. Revoke your change, it is meant to be a procedure. There is a reason setting:get uses .handle, and that is because it returns a value.

Reply:
This is incorrect because analytics themselves are setting settings. Why are you say this?

Reply:
Also setting values should return success or not.